src/middlewared/middlewared/etc_files/local/smbusername.map

```diff
@@ -0,0 +1,21 @@
+#
+# SMB.CONF(5)		The configuration file for the Samba suite 
+#
+<%
+    """
+    The username map is required for proper support of microsoft accounts
+    that are also email addresses. See SMB.CONF(5) for more details.
+    """
+    users = middleware.call_sync('datastore.query', 'account.bsdusers', [
```

Comment:
Shall we use user.query instead of datastore?

Reply:
Yes